src/components/discover-sheet/ListsSection.js

```diff
@@ -0,0 +1,334 @@
+import { keys, times, toLower } from 'lodash';
+import React, {
+  Fragment,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+} from 'react';
+import { FlatList, LayoutAnimation } from 'react-native';
+import { useDispatch, useSelector } from 'react-redux';
+import styled from 'styled-components';
+import { emitAssetRequest, emitChartsRequest } from '../../redux/explorer';
+import { fetchCoingeckoIds } from '../../redux/fallbackExplorer';
+import { DefaultTokenLists } from '../../references';
+import { ButtonPressAnimation } from '../animations';
+import { AssetListItemSkeleton } from '../asset-list';
+import { ListCoinRow } from '../coin-row';
+import { initialChartExpandedStateSheetHeight } from '../expanded-state/ChartExpandedState';
+import { Centered, Column, Flex, Row } from '../layout';
+import { Emoji, Text } from '../text';
+import EdgeFade from './EdgeFade';
+import {
+  useAccountAssets,
+  useAccountSettings,
+  useUserLists,
+} from '@rainbow-me/hooks';
+import { useNavigation } from '@rainbow-me/navigation';
+import Routes from '@rainbow-me/routes';
+import { ethereumUtils } from '@rainbow-me/utils';
+
+const COINGECKO_TRENDING_ENDPOINT =
+  'https://api.coingecko.com/api/v3/search/trending';
+
+const fetchTrendingAddresses = async () => {
+  const trendingAddresses = [];
+  try {
+    const coingeckoIds = await fetchCoingeckoIds();
+    const request = await fetch(COINGECKO_TRENDING_ENDPOINT);
+    const trending = await request.json();
+    const idsToLookUp = trending.coins.map(coin => coin.item.id);
+    keys(coingeckoIds).forEach(address => {
```

Comment:
Hm I think I have an opinion here but I'm not sure what yet - leaving a note for now

Reply:
Following the design patterns we use I guess this should have been its own hook, however, it's not gonna be reused anywhere so I didn't see the need for it.

Reply:
oh yeah code-reuse-wise this is fine! The thing I was wondering was if there was a more efficient lookup pattern for making the trendingAddresses array